### PR TITLE
introduce an extension trait for testing servers

### PR DESCRIPTION
**Pull request by jbr**
This is a first step in the direction of a [superagent](https://visionmedia.github.io/superagent/)/[supertest](https://github.com/visionmedia/supertest) type testing utility for tide. Currently it just supports simple requests and is not exported for usage outside of tide, but the intent is that as it accumulates more features and stabilizes, it might eventually be public under a tide::testing namespace for tide apps to use.

```rust
// this:
    let req = Request::new(
        Method::Get,
        Url::parse("http://example.com/foo/echo").unwrap(),
    );
    let mut res: Response = app.respond(req).await.unwrap();

// becomes:
    let mut res = app.get("/foo/echo").await;
```

and

```rust
 // this:
    let req = http::Request::new(
        Method::Get,
        Url::parse("http://localhost/add_one/3").unwrap(),
    );
    let mut res: http::Response = app.respond(req).await.unwrap();
    assert_eq!(&res.body_string().await.unwrap(), "4");

// becomes:
    assert_eq!(app.get_body("/add_one/3").await, "4");
```

### CODE REVIEW

File: src/endpoint.rs [hunk 1]
```diff
@@ -66,6 +66,12 @@ where
     }
 }
 
+impl<State> Endpoint<State> for &'static str {
```
Comment (Fishrock123) [inline src/endpoint.rs:69]: Why's this in the same change?

Comment (jbr) [reply src/endpoint.rs:69]: See note above ☝️

Comment (Fishrock123) [reply src/endpoint.rs:69]: It'd be nice to have it in a separate commit

Comment (jbr) [reply src/endpoint.rs]: Entirely backed it out, wasn't used anywhere

**Comment by Fishrock123** [comment]
still lgtm

**Pull request merged**